File: src/main/java/de/whitefrog/neobase/model/relationship/Relationship.java
package de.whitefrog.neobase.model.relationship;

import de.whitefrog.neobase.model.Base;

public interface Relationship<From, To> extends Base {
  From getFrom();
  Relationship setFrom(From from);

  To getTo();
  Relationship setTo(To to);
}

File: src/main/kotlin/de/whitefrog/froggy/model/relationship/Relationship.kt
package de.whitefrog.froggy.model.relationship

import de.whitefrog.froggy.model.Base
import de.whitefrog.froggy.model.Model

/**
 * Interface common to all relationship models.
 */
interface Relationship<From : Model, To : Model> : Base {
  var from: From
  var to: To
}
